### PR TITLE
Add better docs for psp clusters

### PR DESCRIPTION
**Pull request by johscheuer**
Document what is required in order to run illuminatio in restricted clusters with psp. 

See: https://github.com/inovex/illuminatio/issues/67

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,6 +174,17 @@ illuminatio 1.1 is confirmed to be working properly with the following kubernete
 - Google Kubernetes Engine, v1.12.8-gke.10
 - kubeadm 1.15.0-00, kubernetes v1.15.2
 
+### PodSecurityPolicy
+
+If your cluster has the [PodSecurityPolcy](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#podsecuritypolicy) Admission Controller you must ensure that illuminatio has the following rights to be created:
```
Comment (maxbischoff) [inline README.md]: Typo:

```suggestion
If your cluster has the [PodSecurityPolicy](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#podsecuritypolicy) Admission Controller you must ensure that illuminatio has the following rights to be created:
```

Comment (maxbischoff) [reply README.md]: Maybe also clarify, that the runner requires those privileges (the CLI doesn't):

```suggestion
If your cluster has the [PodSecurityPolicy](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#podsecuritypolicy) Admission Controller you must ensure that the illuminatio runner has the following rights to be created:
```

File: README.md [hunk 2]
```diff
@@ -174,6 +174,17 @@ illuminatio 1.1 is confirmed to be working properly with the following kubernete
...
+
+- Wants to run as root
+- Needs the `SYS_ADMIN` capability
+- Needs `allowPrivilegeEscalation: true`
+- Needs access to the `hostPath` for the network namespaces and the cri socket
+
+For more details look at the [illuminatio DaemonSet](src/illuminatio/manifests/containerd-daemonset.yaml)
```
Comment (maxbischoff) [inline README.md]: We can at least provide the service account that needs these privileges:

```suggestion
A PodSecurityPolicy granting these privileges needs to be bound to the `illuminatio-runner` service-account in the `illuminatio` namespace.
For more details look at the [illuminatio DaemonSet](src/illuminatio/manifests/containerd-daemonset.yaml)
```

**Comment by johscheuer** [comment]
I updated the docs and fixed the docker cli image to use a different user than root (and some other minor issues)

**Comment by johscheuer** [comment]
I created an issue for providing a PSP -> https://github.com/inovex/illuminatio/issues/73 

IMHO if you run a Cluster with PSP activated you should be able to write it yourself since most tooling in the Kubernetres ecosystem probably doesn't provide them (which doesn't mean we shouldn't :))

File: illuminatio-runner.dockerfile [hunk 3]
```diff
@@ -40,4 +40,4 @@ COPY src /src/app/src
 
 USER runner
```
Comment (maxbischoff) [inline illuminatio-runner.dockerfile]: use nurmeric user here, too? Requires adding `-u 1000` in line 19

File: Dockerfile [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
+FROM python:3.7.4-alpine3.10
+
+RUN adduser -S illuminatio -H -s /bin/nologin -u 1000
+USER 1000
+
+COPY --from=builder /root/.local /home/illuminatio/.local
+ENV PATH=/home/illuminatio/.local/bin:$PATH
 
-CMD ["/bin/sh"]
+ENTRYPOINT [ "illuminatio" ]
```
Comment (maxbischoff) [inline Dockerfile]: Can you also please adjust the [readme section Docker Usage](https://github.com/inovex/illuminatio#docker-usage) to this new entrypoint?

File: illuminatio-runner.dockerfile (hunk 3 above)
Comment (johscheuer) [reply illuminatio-runner.dockerfile]: The user is never used so I wont change it here

File: Dockerfile (hunk 4 above)
Comment (johscheuer) [reply Dockerfile]: Updated

**Comment by johscheuer** [comment]
@maxbischoff  so good you need to approve twice ! 😂

**Pull request merged**